apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -52,6 +52,75 @@ function load_full_site_editing() {
 }
 add_action( 'plugins_loaded', __NAMESPACE__ . '\load_full_site_editing' );
 
+/**
+ * Whether or not FSE is active.
+ * If false, FSE functionality can be disabled.
+ *
+ * @returns bool True if FSE is active, false otherwise.
+ */
+function is_full_site_editing_active() {
+	return is_site_eligible_for_full_site_editing() &&
+			( is_theme_supported() || in_array( get_theme_slug(), SUPPORTED_THEMES, true ) );
+}
+
+/**
+ * Returns normalized theme slug for the current theme.
+ *
+ * Normalize WP.com theme slugs that differ from those that we'll get on self hosted sites.
+ * For example, we will get 'modern-business-wpcom' when retrieving theme slug on self hosted sites,
+ * but due to WP.com setup, on Simple sites we'll get 'pub/modern-business' for the theme.
+ *
+ * @return string Normalized theme slug.
+ */
+function get_theme_slug() {
+	$theme_slug = get_stylesheet();
+	if ( 'pub/' === substr( $theme_slug, 0, 4 ) ) {
+		$theme_slug = substr( $theme_slug, 4 );
+	}
+
+	if ( '-wpcom' === substr( $theme_slug, -6, 6 ) ) {
+		$theme_slug = substr( $theme_slug, 0, -6 );
+	}
+
+	return $theme_slug;
+}
+
+/**
+ * Whether or not the site is eligible for FSE.
+ * This is essentially a feature gate to disable FSE
+ * on some sites which could theoretically otherwise use it.
+ *
+ * @return bool True if current site is eligible for FSE, false otherwise.
+ */
+function is_site_eligible_for_full_site_editing() {
+	/**
+	 * Can be used to disable Full Site Editing functionality.
+	 *
+	 * @since 0.2
+	 *
+	 * @param bool true if Full Site Editing should be disabled, false otherwise.
+	 */
+	return ! apply_filters( 'a8c_disable_full_site_editing', false );
+}
+
+/**
+ * Whether or not current theme is FSE enabled.
+ * This is to allow the checking of theme support to be abstracted
+ * to the context in which the plugin is running.
+ *
+ * @return bool True if current theme supports FSE, false otherwise.
+ */
+function is_theme_supported() {
+	/**
+	 * Can be used to enable Full Site Editing theme support in certian contexts.
+	 *
+	 * @since 0.7
+	 *
+	 * @param bool true if Full Site Editing is available for current theme, false otherwise.
+	 */
+	return apply_filters( 'a8c_fse_enable_theme_support', false );
```

Comment:
Can we fallback to a default function for Jetpack sites, maybe with a high loading number?

Reply:
i think it would be nice to handle the support check here, but get the theme name from the filter instead (that way we don't duplicate the array everywhere)

Reply:
ah that'd make sense too